sdk/cosmos/azure-cosmos-encryption/src/main/java/com/azure/cosmos/encryption/ChangeFeedEncryptionProcessorBuilder.java

```diff
@@ -0,0 +1,129 @@
+
+package com.azure.cosmos.encryption;
+
+import com.azure.cosmos.ChangeFeedProcessor;
+import com.azure.cosmos.ChangeFeedProcessorBuilder;
+import com.azure.cosmos.CosmosAsyncContainer;
+import com.azure.cosmos.models.ChangeFeedProcessorOptions;
+import com.fasterxml.jackson.databind.JsonNode;
+import com.fasterxml.jackson.databind.node.ObjectNode;
+import reactor.core.publisher.Flux;
+import reactor.core.publisher.Mono;
+import reactor.core.scheduler.Schedulers;
+
+import java.util.List;
+import java.util.function.Consumer;
+import java.util.stream.Collectors;
+
+/**
+ * Helper class to build a encryption supported {@link ChangeFeedProcessor} instance.
+ *
+ */
+public class ChangeFeedEncryptionProcessorBuilder {
+
+    private String hostName ;
+    private ChangeFeedProcessorOptions changeFeedProcessorOptions;
+    private Consumer<List<JsonNode>> encryptionConsumer;
+    private CosmosEncryptionAsyncContainer feedContainer = null;
+    private CosmosAsyncContainer leaseContainer = null;
+
+    /**
+     * Helper class to build a encryption supported {@link ChangeFeedProcessor} instance.
+     *
+     */
+    public ChangeFeedEncryptionProcessorBuilder() {
+    }
+
+    /**
+     * Sets the host name.
+     *
+     * @param hostName the name to be used for the host. When using multiple hosts, each host must have a unique
+     * name.
+     * @return current Builder.
+     */
+    public ChangeFeedEncryptionProcessorBuilder hostName(String hostName) {
+        this.hostName = hostName;
+        return this;
+    }
+
+    /**
+     * Sets an existing {@link CosmosEncryptionAsyncContainer} to be used to read from the monitored container.
+     *
+     * @param feedContainer the instance of {@link CosmosEncryptionAsyncContainer} to be used.
+     * @return current Builder.
+     */
+    public ChangeFeedEncryptionProcessorBuilder feedContainer(CosmosEncryptionAsyncContainer feedContainer) {
+        this.feedContainer = feedContainer;
+
+        return this;
+    }
+
+    /**
+     * Sets an existing {@link CosmosAsyncContainer} to be used to read from the leases container.
+     *
+     * @param leaseContainer the instance of {@link CosmosAsyncContainer} to use.
+     * @return current Builder.
+     */
+    public ChangeFeedEncryptionProcessorBuilder leaseContainer(CosmosAsyncContainer leaseContainer) {
+        this.leaseContainer = leaseContainer;
+
+        return this;
+    }
+
+    /**
+     * Sets a consumer function which will be called to process changes.
+     *
+     * @param consumer the {@link Consumer} to call for handling the feeds.
+     * @return current Builder.
+     */
+    public ChangeFeedEncryptionProcessorBuilder handleChanges(Consumer<List<JsonNode>> consumer) {
+        this.encryptionConsumer = jsonNodes -> {
+            List<Mono<byte[]>> byteArrayMonoList =
+                jsonNodes.stream().map(jsonNode -> feedContainer.decryptResponse((ObjectNode) jsonNode)).collect(Collectors.toList());
```

Comment:
ObjectNode objectNode = jsonNode.deepCopy();

https://stackoverflow.com/questions/32713109/how-to-convert-jsonnode-to-objectnode

Reply:
In our case all these are object node , we don't need deepcopy.
 We will revisit logic once we work on this work item of aggregated queries https://github.com/Azure/azure-sdk-for-java/issues/23160

Reply:
Discussed offline, added object node check and throw custom exception if it is not